### PR TITLE
Fix Appointment-bugs

### PR DESCRIPTION
**Pull request by MonishaSivanathan**
- changed the status code as 401 in permission required
- chaned the medical-condition and allergy creation in appointment
- added json response in encounter
- change the regex in patient model

### CODE REVIEW

**Comment by tamsud** [comment]
why are you creating the two different function for saving the single resource (condition/allergy)

**Review by tamsud — CHANGES_REQUESTED** [review]

File: controller/appointment_controller.py [hunk 1]
```diff
@@ -795,3 +820,115 @@ def update_appointment_status(update_status: StatusModel, appointment_id: str):
                 content=error_response_data,
                 status_code=status.HTTP_400_BAD_REQUEST
             )
+
+
+    @staticmethod
+    def create_allergy_app(appointment, patient_id, allergy, values): 
+        appointment_result = AllergyIntolerance(
```
[the remark this answers is not in the document]
Comment (MonishaSivanathan) [reply controller/appointment_controller.py]: ok and changed

File: controller/appointment_controller.py [hunk 2]
```diff
@@ -795,3 +820,115 @@ def update_appointment_status(update_status: StatusModel, appointment_id: str):
...
+                coding=[Coding(system="http://terminology.hl7.org/CodeSystem/allergyintolerance-verification", code="confirmed")],
+                text=appointment.id
+            )
+        )
+        appointment_result.save()
+        return appointment_result.id
+
+    @staticmethod
+    def update_allergy_app(appointment, patient_id, allergy, values, current_allergy_id):
+        response_condition = Condition.make_request(
```
[the remark this answers is not in the document]
Comment (MonishaSivanathan) [reply controller/appointment_controller.py]: changed to allergy

File: controller/appointment_controller.py [hunk 3]
```diff
@@ -795,3 +820,115 @@ def update_appointment_status(update_status: StatusModel, appointment_id: str):
...
+                coding=[Coding(system="http://terminology.hl7.org/CodeSystem/condition-verification", code="confirmed")],
+                text=appointment.id
+            )
+        )
+        additional_condition.save()
+        return additional_condition.id
+    
+    @staticmethod
+    def update_condition_app(appointment, patient_id, condition, condition_id, values):
+        response_allergy = Condition.make_request(
```
[the remark this answers is not in the document]
Comment (MonishaSivanathan) [reply controller/appointment_controller.py]: changed

File: controller/appointment_controller.py [hunk 4]
```diff
@@ -142,6 +145,9 @@ def create_appointment(patient_id: str, app: AppoinmentModel):
                     response_data["current_condition_id"] = current_condition.id
                     response_data["current_condition"] = "Created"
 
+                    current_condition_app_data = AppointmentClient.create_update_condition(appointment.id, patient_id, app.current_medical_condition, CURRENT)
```
Comment (tamsud) [inline controller/appointment_controller.py:148]: you have current condition id here why are you not passing it on the function?

File: controller/appointment_controller.py [hunk 5]
```diff
@@ -180,6 +189,9 @@ def create_appointment(patient_id: str, app: AppoinmentModel):
                     response_data["other_condition_id"] = additional_condition.id
                     response_data["other_condition"] = "Created"
 
+                    additional_condition_app_data = AppointmentClient.create_update_condition(appointment.id, patient_id, app.other_medical_condition, OTHER)
```
Comment (tamsud) [inline controller/appointment_controller.py:192]: why additional_condition.id is not passed to the fucntion?

File: controller/appointment_controller.py [hunk 6]
```diff
@@ -218,6 +233,9 @@ def create_appointment(patient_id: str, app: AppoinmentModel):
                     response_data["current_allergy_id"] = current_allergy.id
                     response_data["current_allergy"] = "Created"
 
+                    current_allergy_app_data = AppointmentClient.create_update_allergy(appointment.id, patient_id, app.current_allergy, CURRENT)
```
Comment (tamsud) [inline controller/appointment_controller.py:236]: same here, id is missing?

File: controller/appointment_controller.py [hunk 7]
```diff
@@ -255,6 +276,9 @@ def create_appointment(patient_id: str, app: AppoinmentModel):
                     additional_allergy.save()
                     response_data["other_allergy_id"] = additional_allergy.id
                     response_data["other_allergy"] = "Created"
+
+                    other_allergy_app_data = AppointmentClient.create_update_allergy(appointment.id, patient_id, app.current_allergy, OTHER)
```
Comment (tamsud) [inline controller/appointment_controller.py:280]: id is missing?

File: controller/appointment_controller.py [hunk 8]
```diff
@@ -795,3 +820,115 @@ def update_appointment_status(update_status: StatusModel, appointment_id: str):
...
+                method="GET", endpoint=f"/fhir/AllergyIntolerance/{allergy_id}"
+            )
+            condition_data = response_condition.json()
+            verification_status = False
+            for entry in condition_data.get("entry", []):
+                if entry["resource"].get("verificationStatus"):
+                    verification_status = True
+            
+            if verification_status:
+                return allergy_id
```
Comment (tamsud) [inline controller/appointment_controller.py:838]: - Why are you returning the ID immediately? What's the reason behind it? you always passing a default ID right?
- Why is this check necessary here?
- What would happen if there are multiple appointments for the same patient?

File: controller/appointment_controller.py [hunk 9]
```diff
@@ -795,3 +820,115 @@ def update_appointment_status(update_status: StatusModel, appointment_id: str):
...
+                    text=appointment_id
+                )
+            )
+        allergy_instance.save()
+        return allergy_instance.id
+
+    
+    @staticmethod
+    def create_update_condition(appointment_id, patient_id, condition, values, condition_id=None):
+        if condition_id:
```
Comment (tamsud) [inline controller/appointment_controller.py:883]: - Why are you returning the ID immediately? What's the reason behind it? you always passing a default ID right?
- Why is this check necessary here?
- What would happen if there are multiple appointments for the same patient?

**Review by tamsud — CHANGES_REQUESTED** [review]

File: controller/appointment_controller.py (hunk 4 above)
Comment (MonishaSivanathan) [reply controller/appointment_controller.py]: changed according to the new screen

File: controller/appointment_controller.py (hunk 6 above)
Comment (MonishaSivanathan) [reply controller/appointment_controller.py]: changed according to the new screen

File: controller/appointment_controller.py (hunk 7 above)
Comment (MonishaSivanathan) [reply controller/appointment_controller.py]: changed according to the new screen

File: controller/appointment_controller.py (hunk 8 above)
Comment (MonishaSivanathan) [reply controller/appointment_controller.py]: changed according to the new screen

File: controller/appointment_controller.py (hunk 9 above)
Comment (MonishaSivanathan) [reply controller/appointment_controller.py]: changed according to the new screen

**Review by tamsud — APPROVED** [review]

**Pull request merged**